vendor/github.com/jszwedko/go-circleci/circleci.go

```diff
@@ -11,19 +11,30 @@ import (
 	"net/http/httputil"
 	"net/url"
 	"os"
+	"regexp"
 	"strconv"
 	"time"
 )
 
 const (
-	queryLimit = 100 // maximum that CircleCI allows
+	queryLimit        = 100                         // maximum that CircleCI allows
+	envVarNameValidRE = "^[[:alpha:]]+[[:word:]]*$" // source https://circleci.com/docs/2.0/env-vars/#injecting-environment-variables-with-the-api
 )
 
 var (
-	defaultBaseURL = &url.URL{Host: "circleci.com", Scheme: "https", Path: "/api/v1.1/"}
-	defaultLogger  = log.New(os.Stderr, "", log.LstdFlags)
+	defaultBaseURL            = &url.URL{Host: "circleci.com", Scheme: "https", Path: "/api/v1.1/"}
+	defaultLogger             = log.New(os.Stderr, "", log.LstdFlags)
+	envVarNameValidCompiledRE *regexp.Regexp
 )
 
+func init() {
+	compiledRE, err := regexp.Compile(envVarNameValidRE)
```

Comment:
This has nothing to do with the provider code I know, but I would use [`regexp.MustCompile`](https://golang.org/pkg/regexp/#MustCompile)  here and remove the need to have an `init` func at all.

Reply:
Indeed, if I just panic to handle the error, better use `MustCompile` directly which has the same behavior. I'll change it.